rfcs/rfc-net-20211026-1-flexmatch.md

```diff
@@ -0,0 +1,221 @@
+ 
+#### Client APIs
+ 
+Update [`ISessionRequests.h`](https://github.com/o3de/o3de/blob/development/Code/Framework/AzFramework/AzFramework/Session/ISessionRequests.h) by adding client public API requests.
+ 
+```
+struct AcceptMatchRequest
+{
+    bool m_acceptMatch;
+    AZStd::vector<AZStd::string> m_playerIds;
+    AZStd::string m_ticketId;
+};
+ 
+struct StartMatchmakingRequest
+{
+    AZStd::string m_ticketId;
+};
+ 
+struct StopMatchmakingRequest
+{
+    AZStd::string m_ticketId;
+};
+```
+ 
+Update [`ISessionRequests`](https://github.com/o3de/o3de/blob/development/Code/Framework/AzFramework/AzFramework/Session/ISessionRequests.h#L107) for `AcceptMatch`, `StartMatchmaking` and `StopMatchmaking`
+ 
+```
+virtual void AcceptMatch(const AcceptMatchRequest& request) = 0;
+ 
+virtual AZStd::string StartMatchmaking(const StartMatchmakingRequest& request) = 0;
+ 
+virtual void StopMatchmaking(const StopMatchmakingRequest& request) = 0;
+```
+ 
+Update [`ISessionAsyncRequests`](https://github.com/o3de/o3de/blob/development/Code/Framework/AzFramework/AzFramework/Session/ISessionRequests.h#L136) and [`SessionAsyncRequestNotifications`](https://github.com/o3de/o3de/blob/development/Code/Framework/AzFramework/AzFramework/Session/ISessionRequests.h#L162) for `AcceptMatch`, `StartMatchmaking` and `StopMatchmaking`
+ 
+```
+virtual void AcceptMatchAsync(const AcceptMatchRequest& request) = 0;
+virtual void OnAcceptMatchAsyncComplete() = 0;
+ 
+virtual void StartMatchmakingAsync(const StartMatchmakingRequest& request) = 0;
+virtual void OnStartMatchmakingAsyncComplete(const AZStd::string& matchmakingTicketId) = 0;
+ 
+virtual void StopMatchmakingAsync(const StopMatchmakingRequest& request) = 0;
+virtual void OnStopMatchmakingAsyncComplete() = 0;
+```
+ 
+Add AWS GameLift Gem matchmaking requests `AWSGameLiftAcceptMatchRequest`, `AWSGameLiftStartMatchmakingRequest` and `AWSGameLiftStopMatchmakingRequest` under [`Gems/AWSGameLift/Code/AWSGameLiftClient/Include/Request`](https://github.com/o3de/o3de/tree/development/Gems/AWSGameLift/Code/AWSGameLiftClient/Include/Request) as Amazon GameLift service requires extra attributes for matchmaking request [6](https://docs.aws.amazon.com/gamelift/latest/apireference/API_StartMatchmaking.html)
+ 
+#### Server Notifications
+ 
+Update [`SessionNotifications`](https://github.com/o3de/o3de/blob/development/Code/Framework/AzFramework/AzFramework/Session/SessionNotifications.h) with `OnUpdateSessionBegin`
+ 
+```
+virtual void OnUpdateSessionBegin(const SessionConfig& sessionConfig) = 0;
+```
+ 
+#### Server APIs
+ 
+Update [`IAWSGameLiftServerRequests`](https://github.com/o3de/o3de/blob/development/Gems/AWSGameLift/Code/AWSGameLiftServer/Include/Request/IAWSGameLiftServerRequests.h) with backfill server APIs
+ 
+```
+class IAWSGameLiftServerRequests
+{
+public:
+    virtual void StartMatchBackfill(const StartMatchBackfillRequest& request) = 0;
```

Comment:
Minor update:
```
virtual bool StartMatchBackfill(const AZStd::string& ticketId, const AZStd::vector<AWSGameLift::AWSGameLiftPlayer>& players) = 0;

virtual bool StopMatchBackfill(const AZStd::string& ticketId) = 0;
```

Reply:
We don't have a clear policy for updating RFCs. IMHO APIs in RFCs are suggestions of the APIs, the actual PRs related to the story should take to changes required for delivery.